cdap-common/bin/functions.sh

```diff
@@ -413,7 +415,22 @@ cdap_set_hbase() {
     1.2-cdh*) __compat="${CDAP_HOME}"/hbase-compat-1.2-cdh5.7.0/lib/* ;; # 5.7 and 5.8 are compatible
     1.2*) __compat="${CDAP_HOME}"/hbase-compat-1.1/lib/* ;; # 1.1 and 1.2 are compatible
     "") die "Unable to determine HBase version! Aborting." ;;
-    *) die "Unknown/Unsupported HBase version found: ${HBASE_VERSION}" ;;
+    *-cdh*)
+      if [[ $(cdap_get_conf "hbase.version.resolution.strategy" "${CDAP_CONF}"/cdap-site.xml auto.strict) == 'auto.latest' ]]; then
+        __compat="${CDAP_HOME}"/"${__latest_hbase_compat_cdh}"/lib/* # if allowed, return the latest CDH HBase version for unsupported HBase version
+        echo "'auto.latest' resolution strategy found. Using ${__latest_hbase_compat_cdh} for HBase version: ${HBASE_VERSION}"
+      else
+        die "Unknown or Unsupported CDH HBase version found: ${HBASE_VERSION}"
+      fi
+      ;;
+    *)
+      if [[ $(cdap_get_conf "hbase.version.resolution.strategy" "${CDAP_CONF}"/cdap-site.xml auto.strict) == 'auto.latest' ]]; then
+        __compat="${CDAP_HOME}"/"${__latest_hbase_compat}"//lib/* # if allowed, return the latest HBase version for unsupported HBase version
+        echo "'auto.latest' resolution strategy found. Using ${__latest_hbase_compat_cdh} for HBase version: ${HBASE_VERSION}"
+      else
+        die "Unknown or Unsupported HBase version found: ${HBASE_VERSION}"
+      fi
+      ;;
```

Comment:
These two case statements are basically identical. Rather than set these strings to single-use variables, you should use an actually meaningful variable. First, drop the variables at L405 and L406. Then, replace this entire code block with something more concise, putting them both into a single block. This will DRY up the code.
```
    *)
      if [[ $(cdap_get_conf "hbase.version.resolution.strategy" "${CDAP_CONF}"/cdap-site.xml auto.strict) == 'auto.latest' ]]; then
        if [[ ${HBASE_VERSION} =~ -cdh ]]; then
          local readonly __latest_hbase_compat=hbase-compat-1.2-cdh5.7.0
        else
          local readonly __latest_hbase_compat=hbase-compat-1.1
        fi
        __compat="${CDAP_HOME}"/${__latest_hbase_compat}/lib/*
        echo "Using ${__latest_hbase_compat} for HBase version ${HBASE_VERSION} due to 'auto.latest' resolution strategy."
      else
        die "Unknown or unsupported HBase version found: ${HBASE_VERSION}"
      fi
      ;;
```